### ISSUE TITLE
OpenIdConnect events called twice for a single login - when using EnableTokenAcquisitionToCallDownstreamApi 

### ISSUE THREAD

**Issue by markharrison** [AzureAD/microsoft-identity-web#2083]
### Microsoft.Identity.Web Library

Microsoft.Identity.Web

### Microsoft.Identity.Web version

1.26.0

### Web app

Sign-in users

### Web API

Not Applicable

### Token cache serialization

In-memory caches

### Description

When I add 
```
     .EnableTokenAcquisitionToCallDownstreamApi(builder.Configuration.GetValue<string>
                            ("DownstreamApi:Scopes")?.Split(' '))
     .AddInMemoryTokenCaches();
```
to my startup logic - then during my login activity, all my OpenId events are called twice.

Why is this happening?

### Reproduction steps

Use Code A - get events called twice for a single logon

Use Code B - get events called once for a single logon

using Framework : .NET 7.0.3

### Error message

n/a

### Id Web logs

I have log /trace messages when the event is called

Logs A - events appear twice for single login
```
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.46.0.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22621 [2023-02-15 17:47:14Z - 9d18da67-e9a5-4e35-a36f-5b962176077b] Saving Id Token and Account in cache ...
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.46.0.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22621 [2023-02-15 17:47:14Z - 9d18da67-e9a5-4e35-a36f-5b962176077b] Saving RT in cache...
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.46.0.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22621 [2023-02-15 17:47:14Z - 9d18da67-e9a5-4e35-a36f-5b962176077b] 
	=== Token Acquisition finished successfully:
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.46.0.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22621 [2023-02-15 17:47:14Z - 9d18da67-e9a5-4e35-a36f-5b962176077b]  AT expiration time: 15/02/2023 19:02:24 +00:00, scopes: openid profile User.Read email. source: IdentityProvider
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.46.0.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22621 [2023-02-15 17:47:14Z - 9d18da67-e9a5-4e35-a36f-5b962176077b] Fetched access token from host login.microsoftonline.com. 
Program: Information: OnAuthorizationCodeReceivedFunc
Program: Information: OnAuthorizationCodeReceivedFunc
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Dev\IdentityWeb\bin\Debug\net7.0\Microsoft.IdentityModel.JsonWebTokens.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: Microsoft.IdentityModel Version: 6.25.1.0. Date 02/15/2023 17:47:14. PII logging is OFF. See https://aka.ms/IdentityModel/PII for details. 
IDX10242: Security token: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' has a valid signature.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10239: Lifetime of the token is valid.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10234: Audience Validated.Audience: '8bfd421f-9888-4e8c-b9c4-504b664b2b74'
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10245: Creating claims identity from the validated token: '[PII of type 'System.IdentityModel.Tokens.Jwt.JwtSecurityToken' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10241: Security token validated. token: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
Program: Information: OnTokenValidatedFunc
Program: Information: OnTokenValidatedFunc
Program: Information: OnTicketReceivedFunc
Program: Information: OnTicketReceivedFunc
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.46.0.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22621 [2023-02-15 17:47:14Z - f05273b4-a20c-496b-9057-f9c79a897918] [Region discovery] Not using a regional authority. 
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.46.0.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22621 [2023-02-15 17:47:14Z - f05273b4-a20c-496b-9057-f9c79a897918] [Region discovery] Not using a regional authority. 
Microsoft.Identity.Web.TokenAcquisition: In
```

Logs B 
```
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.3\System.Security.Cryptography.Algorithms.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.3\System.Text.RegularExpressions.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Program: Information: OnRedirectToIdentityProviderFunc
Program: Information: OnAuthorizationCodeReceivedFunc
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Dev\IdentityWeb\bin\Debug\net7.0\Microsoft.IdentityModel.JsonWebTokens.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: Microsoft.IdentityModel Version: 6.25.1.0. Date 02/15/2023 17:53:50. PII logging is OFF. See https://aka.ms/IdentityModel/PII for details. 
IDX10242: Security token: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' has a valid signature.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10239: Lifetime of the token is valid.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10234: Audience Validated.Audience: '8bfd421f-9888-4e8c-b9c4-504b664b2b74'
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10245: Creating claims identity from the validated token: '[PII of type 'System.IdentityModel.Tokens.Jwt.JwtSecurityToken' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10241: Security token validated. token: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
Program: Information: OnTokenValidatedFunc
Program: Information: OnUserInformationReceivedFunc
'iisexpress.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\Microsoft Visual Studio\2022\Community\Common7\IDE\PrivateAssemblies\Runtime\Microsoft.VisualStudio.Debugger.Runtime.NetCoreApp.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
```

### Relevant code snippets

```csharp
Code A:

builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
      .AddMicrosoftIdentityWebApp(options =>
      {
          builder.Configuration.Bind("AzureAD", options);
          options.ResponseType = OpenIdConnectResponseType.Code;
          options.Events ??= new OpenIdConnectEvents();
          options.Events.OnTokenValidated += OnTokenValidatedFunc;
          options.Events.OnAuthenticationFailed += OnAuthenticationFailedFunc;
          options.Events.OnAccessDenied += OnAccessDeniedFunc;
          options.Events.OnRemoteFailure += OnRemoteFailureFunc;
          options.Events.OnUserInformationReceived += OnUserInformationReceivedFunc;
          options.Events.OnRedirectToIdentityProvider += OnRedirectToIdentityProviderFunc;
          options.Events.OnTicketReceived += OnTicketReceivedFunc;
          options.Events.OnAuthorizationCodeReceived += OnAuthorizationCodeReceivedFunc;
          options.SaveTokens = true;
          options.UsePkce = true;
          options.GetClaimsFromUserInfoEndpoint = true;
          options.TokenValidationParameters = new TokenValidationParameters
          {
              RoleClaimType = "role",
              NameClaimType = "name"
          };
      }
    )
     .EnableTokenAcquisitionToCallDownstreamApi(builder.Configuration.GetValue<string>
                                            ("DownstreamApi:Scopes")?.Split(' '))
     .AddInMemoryTokenCaches();

```
Code B:
```
builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
      .AddMicrosoftIdentityWebApp(options =>
      {
          builder.Configuration.Bind("AzureAD", options);
          options.ResponseType = OpenIdConnectResponseType.Code;
          options.Events ??= new OpenIdConnectEvents();
          options.Events.OnTokenValidated += OnTokenValidatedFunc;
          options.Events.OnAuthenticationFailed += OnAuthenticationFailedFunc;
          options.Events.OnAccessDenied += OnAccessDeniedFunc;
          options.Events.OnRemoteFailure += OnRemoteFailureFunc;
          options.Events.OnUserInformationReceived += OnUserInformationReceivedFunc;
          options.Events.OnRedirectToIdentityProvider += OnRedirectToIdentityProviderFunc;
          options.Events.OnTicketReceived += OnTicketReceivedFunc;
          options.Events.OnAuthorizationCodeReceived += OnAuthorizationCodeReceivedFunc;
          options.SaveTokens = true;
          options.UsePkce = true;
          options.GetClaimsFromUserInfoEndpoint = true;
          options.TokenValidationParameters = new TokenValidationParameters
          {
              RoleClaimType = "role",
              NameClaimType = "name"
          };
      }
    );
```
```


### Regression

_No response_

### Expected behavior

Events called once for a single login

**Comment by markharrison** [comment]
im still seeing the same issue with v2.5 - the hooks are called twice 

Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.48.1.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22623 [2023-03-02 22:17:20Z - 910755aa-16aa-4be7-b6e4-3180415dfa3e] TokenEndpoint: ****
**Program: Information: OnAuthorizationCodeReceivedFunc
Program: Information: OnAuthorizationCodeReceivedFunc**
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10242: Security token: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]' has a valid signature.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10239: Lifetime of the token is valid.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10234: Audience Validated.Audience: '4a367929-16e9-4882-81e0-18d4e1a1479b'
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10245: Creating claims identity from the validated token: '[PII of type 'System.IdentityModel.Tokens.Jwt.JwtSecurityToken' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
Microsoft.IdentityModel.LoggingExtensions.IdentityLoggerAdapter: Information: IDX10241: Security token validated. token: '[PII of type 'System.String' is hidden. For more details, see https://aka.ms/IdentityModel/PII.]'.
**Program: Information: OnTokenValidatedFunc
Program: Information: OnTokenValidatedFunc
Program: Information: OnUserInformationReceivedFunc
Program: Information: OnUserInformationReceivedFunc
Program: Information: OnTicketReceivedFunc
Program: Information: OnTicketReceivedFunc**
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.48.1.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22623 [2023-03-02 22:17:22Z] [Internal cache] Total number of cache partitions found while getting refresh tokens: 2
Microsoft.Identity.Web.TokenAcquisition: Information: False MSAL 4.48.1.0 MSAL.NetCore .NET 7.0.3 Microsoft Windows 10.0.22623 [2023-03-02 22:17:22Z - 390d899f-eb38-4431-ba50-5375f39b8516] [Region discovery] Not using a regional authority.

**Comment by markharrison** [comment]
this is my startup logic 

```c#
builder.Services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
      .AddMicrosoftIdentityWebApp(options =>
      {
          builder.Configuration.Bind("AzureAD", options);
          options.ResponseType = OpenIdConnectResponseType.Code;
          options.Events ??= new OpenIdConnectEvents();
          options.Events.OnTokenValidated += OnTokenValidatedFunc;
          options.Events.OnAuthenticationFailed += OnAuthenticationFailedFunc;
          options.Events.OnAccessDenied += OnAccessDeniedFunc;
          options.Events.OnRemoteFailure += OnRemoteFailureFunc;
          options.Events.OnUserInformationReceived += OnUserInformationReceivedFunc;
          options.Events.OnRedirectToIdentityProvider += OnRedirectToIdentityProviderFunc;
          options.Events.OnTicketReceived += OnTicketReceivedFunc;
          options.Events.OnAuthorizationCodeReceived += OnAuthorizationCodeReceivedFunc;
          options.SaveTokens = true;
          options.UsePkce = true;
          options.GetClaimsFromUserInfoEndpoint = true;
          options.TokenValidationParameters = new TokenValidationParameters
          {
              RoleClaimType = "role",
              NameClaimType = "name"
          };
      }
    )
     .EnableTokenAcquisitionToCallDownstreamApi(builder.Configuration.GetValue<string>("ScopesGraph")?.Split(' '))
     .AddInMemoryTokenCaches();
```

**Comment by jmprieur** [comment]
@markharrison we think it's fixed in 2.7.0. Did you try it?

**Comment by markharrison** [comment]
Unfortunately - its still doing it 

![mark](https://user-images.githubusercontent.com/5177824/229495316-8e7f13e5-7065-4d0b-aa98-e95d353019d4.png)

**Comment by jmprieur** [comment]
@markharrison : this should be fixed in 2.13.2
Feel free to reopen if this still repro-es
